worlds/pokemon_rb/__init__.py

```diff
@@ -209,196 +225,377 @@ def encode_name(name, t):
         # to the way effectiveness messages are generated.
         self.type_chart = sorted(chart, key=lambda matchup: -matchup[2])
-        self.multiworld.early_items[self.player]["Exp. All"] = 1
-
-    def create_items(self) -> None:
-        start_inventory = self.multiworld.start_inventory[self.player].value.copy()
-        if self.multiworld.randomize_pokedex[self.player] == "start_with":
-            start_inventory["Pokedex"] = 1
-            self.multiworld.push_precollected(self.create_item("Pokedex"))
 
-        locations = [location for location in location_data if location.type == "Item"]
-        item_pool = []
-        combined_traps = (self.multiworld.poison_trap_weight[self.player].value
-                          + self.multiworld.fire_trap_weight[self.player].value
-                          + self.multiworld.paralyze_trap_weight[self.player].value
-                          + self.multiworld.ice_trap_weight[self.player].value)
-        for location in locations:
-            event = location.event
-            if not location.inclusion(self.multiworld, self.player):
-                continue
-            if location.original_item in self.multiworld.start_inventory[self.player].value and \
-                    location.original_item in item_groups["Unique"]:
-                start_inventory[location.original_item] -= 1
-                item = self.create_filler()
-            elif location.original_item is None:
-                item = self.create_filler()
-            elif location.original_item == "Pokedex":
-                if self.multiworld.randomize_pokedex[self.player] == "vanilla":
-                    self.multiworld.get_location(location.name, self.player).event = True
-                    event = True
-                item = self.create_item("Pokedex")
-            elif location.original_item.startswith("TM"):
-                if self.multiworld.randomize_tm_moves[self.player]:
-                    item = self.create_item(location.original_item.split(" ")[0])
-                else:
-                    item = self.create_item(location.original_item)
-            else:
-                item = self.create_item(location.original_item)
-                if (item.classification == ItemClassification.filler and self.multiworld.random.randint(1, 100)
-                        <= self.multiworld.trap_percentage[self.player].value and combined_traps != 0):
-                    item = self.create_item(self.select_trap())
-            if event:
-                self.multiworld.get_location(location.name, self.player).place_locked_item(item)
-            elif "Badge" not in item.name or self.multiworld.badgesanity[self.player].value:
-                item_pool.append(item)
+        self.dexsanity_table = [
+            *[True for _ in range(round(self.multiworld.dexsanity[self.player].value * 1.51))],
+            *[False for _ in range(151 - round(self.multiworld.dexsanity[self.player].value * 1.51))]
```

Comment:
```suggestion
            *(True for _ in range(round(self.multiworld.dexsanity[self.player].value * 1.51))),
            *(False for _ in range(151 - round(self.multiworld.dexsanity[self.player].value * 1.51)))
```

If they don't persist, they should be generators.